### ISSUE TITLE
lint: add flag-deref checker

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#162]
Detects invalid usages of `flag` package.

**Before:**
```go
a := *flag.Bool("a", false, "a bool flag")
```

**After:**
```go
var a bool
flag.BoolVar(&a, false, "a bool flag")
```

**Comment by cristaloleg** [comment]
Why it’s an invalid usage?

**Comment by Quasilyte** [comment]
Because if you deref pointer returned by that call, value will not be updated after call to `flag.Parse`.

**Comment by Quasilyte** [comment]
You either hold a pointer, or use `XxxVar` flag versions to store flags into existing values, this way you avoid annoying pointers all over the place.
